### PR TITLE
(feat) O3-1942: Forms in workspace should be filtered by user privileges

### PR DESCRIPTION
**Pull request by Jexsie**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [x] My work includes tests or is validated by existing tests.

## Summary

 - Each encounter type has associated user privileges (viewPrivilege and editPrivilege. Previously, we used these to filter 
    forms in the [forms component](https://github.com/openmrs/openmrs-esm-patient-chart/blob/40a0b56f6bf6905061a24a93225e6353c0a9d874/packages/esm-patient-forms-app/src/forms/forms.component.tsx#L40); 
   Lifted this logic to be handled directly from the **useForms** hook with the offline awareness logic
## Related Issue

https://issues.openmrs.org/browse/O3-1942

### CODE REVIEW

**Comment by Jexsie** [comment]
cc @ibacher @denniskigen

File: packages/esm-patient-forms-app/src/hooks/use-forms.ts [hunk 1]
```diff
@@ -43,16 +50,24 @@ export function useForms(patientUuid: string, startDate?: Date, endDate?: Date,
   const encountersRes = useEncountersWithFormRef(patientUuid, startDate, endDate);
   const pastEncounters = encountersRes.data?.data?.results ?? [];
   const data = allFormsRes.data ? mapToFormCompletedInfo(allFormsRes.data, pastEncounters) : undefined;
+  const { htmlFormEntryForms } = useConfig() as ConfigObject;
```
Comment (ibacher) [inline packages/esm-patient-forms-app/src/hooks/use-forms.ts:53]:
```suggestion
  const { htmlFormEntryForms } = useConfig<ConfigObject>();
```

File: packages/esm-patient-forms-app/src/forms/forms-list.test.tsx [hunk 2]
```diff
@@ -10,6 +10,15 @@ import { openmrsFetch, useConfig } from '@openmrs/esm-framework';
 const mockedOpenmrsFetch = openmrsFetch as jest.Mock;
 const mockedUseConfig = useConfig as jest.Mock;
 
+jest.mock('@openmrs/esm-framework', () => {
+  const originalModule = jest.requireActual('@openmrs/esm-framework');
+
+  return {
+    ...originalModule,
+    userHasAccess: jest.fn().mockImplementation(() => true),
```
Comment (ibacher) [inline packages/esm-patient-forms-app/src/forms/forms-list.test.tsx:18]: It'd be nice if we had a test confirming this functions...

File: packages/esm-patient-forms-app/src/hooks/use-forms.ts (hunk 1 above)
Comment (ibacher) [reply packages/esm-patient-forms-app/src/hooks/use-forms.ts:53]: Oh... whoops, yeah, we'd need to update the version of core. I guess it's fine for now.

**Pull request closed without merging**